docs/guides/chip_tool_guide.md

```diff
@@ -1391,3 +1391,180 @@ $ ./chip-tool any read-by-id <cluster-ids> <attribute-ids> <destination-id> <end
+
+For security reasons, all communication between users and credentials happens only using their respective index values. No other information is shared between both pools.
+
+The CHIP Tool lets you add users and credentials on door lock devices and securely match their indexes to one another. This is an optional feature available only when working with devices that implement the `doorlock` cluster.
+
+> **Note:**
+> Users and credentials can only be modified by whoever has the right permissions, as specified in the Access Control List.
+
+Complete the following steps:
+
+1. Set up a user on the device.
+1. Assign a credential for the newly created user.
+
+### Step 1: Set up a user
+
+To set up a user on a door lock device with the CHIP Tool, use the following command pattern:
+
+```
+$ ./chip-tool doorlock set-user <OperationType> <UserIndex> <UserName> <UserUniqueID> <UserStatus> <UserType> <CredentialRule> <destination-id> <endpoint-id> --timedInteractionTimeoutMs 1000
+```
+
+In this command:
+
+-   _<OperationType\>_ is one of the available types of operation for the user:
+
+    - `Add` - This operation adds a new user to the slot at _<UserIndex\>_.
+    - `Clear` - This operation removes an existing user from the slot at _<UserIndex\>_.
+    - `Modify` - This operation modifies an existing user at the slot at _<UserIndex\>_.
+
+-   _<UserIndex\>_ is the index value of the user, between `1` and the maximum value you set in the `NumberOfTotalUsersSupported` attribute. Setting the user index to `0` will cause an error.
+-   _<UserName\>_ is the name of the user, which can have maximum 10 bytes of size.
+-   _<UserUniqueID\>_ is a 4-byte number that describes the unique user ID.
+-   _<UserStatus\>_ can be set to one of the following values:
+
+    - Available (`0`) - This status indicates that the given user slot is available for modification on the device.
+    - Occupied (`1`) - This status indicates that the given user slot is used and active.
+
+-   _<CredentialRule\>_ is the number of credentials that must be used to unlock the door lock. This parameter can have the following values:
+
+    - `0` (Single) - One credential type is required to unlock.
+    - `1` (Dual) - Two credential types are required to unlock.
+    - `2` (Tri) - Three credential types are required to unlock.
+
+-   _<destination-id\>_ is the ID of the door lock device.
+-   _<endpoint-id\>_ is the ID of the endpoint on the door lock device.
+-   `--timedInteractionTimeoutMs` is the timeout duration for the interaction, in milliseconds. This should allow enough time for the interaction to complete.
+
+**Example of command:**
+
+```
+$ ./chip-tool doorlock set-user 0 1 AAA 6452 1 0 0 1 1 --timedInteractionTimeoutMs 1000
```

Comment:
The simplest way to add a user is to SetUser with Add, the index, and everything else null....